airbytehq/airbyte
# Java CDK updates for mysql

ported from https://github.com/airbytehq/airbyte/pull/34608 to kotlin.

* TestDatabase
    * general cleanup (nullability, etc)
    * `ConfigBuilder.build()` now declares ObjectNode as its return type for convenience
* AbstractJdbcDestination - allow injecting a custom v1v2 migrator
* JdbcDestinationHandler
    * Make databaseName nullable (see comment for reason)
    * allow overriding the data type for `_airbyte_destination_state.updated_at` (mysql doesn't have a timestamp_with_timezone type)
    * allow overriding a few DB metadata-related things (mysql doesn't have schemas, so need to pass `namespace` into the `database` argument)
    * type name comparison now uses  `.equals(..., ignoreCase=true)`
* JdbcSqlGenerator
    * add a `formatTimestampLiteral` method
    * kill the createIndexes method - implementing classes can just return a `Sql` object with multiple statements
    * use dslContext everywhere instead of static DSL
    * allow overriding `renameTable` and `createV2RawTableFromV1Table` (once again, to support mysql quirks)
* JdbcV1V2Migrator - more `open` stuff, make `databaseName` nullable
* JdbcSqlGeneratorIntegrationTest - nullability, make `dslContext` protected instead of private
* JdbcTypingDedupingTest
    * make more things open/protected
    * call `convertStreamName` on the raw table namespace (this is to match the JdbcSqlGenerator behavior, so that tests can correctly find the raw table)
    * more DSL -> dslContext changes
* JdbcBufferedConsumerFactory - use ParsedCatalog as source of truth for raw table name
* JdbcSqlOperations: modify airbyte_meta handling (write explicit value when null)
* CatalogParser - correctly propagate the original stream name+namespace when there are table name collisions
* BaseTypingDedupingTest
    * add more tests/assertions
* misc code cleanup
    * lots of making things nonnullable
